core-default/src/main/java/me/aurium/beetle/defaults/task/CommonTaskRunner.java

```diff
@@ -23,6 +24,10 @@ public void executeSync(Runnable runnable) {
             runnable.run();
 
             return null;
+        }).whenComplete((ignored,exception) -> {
+            if (exception != null) {
+                throw new TaskFutureException(exception);
```

Comment:
This still swallows the exception. The issue is that, if a computation inside one of these lambdas throws an exception, the exception is stored inside the future. If you rethrow the exception as you do here, you're still having that exception be stored in the future.

To fix this, you have to log the exception using one of the following:
```java
future.exceptionally((ex) -> {
  LOGGER.warn("Encountered exception", ex);
  return null;
});

future.whenComplete((ignored, ex) -> {
  if (ex != null) {
    LOGGER.warn("Encountered exception", ex);
  }
});
```
Theoretically, if Logger#warn throws an exception, then *that exception* will be swallowed. But you don't have to worry about that in practice.

Reply:
Ah, that's incredibly annoying. I completely forgot about that, haha.